src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/dataobject.cs

```diff
@@ -2863,13 +2863,29 @@ private object GetDataFromHGLOBAL(string format, IntPtr hglobal)
                     {
                         data = ReadBitmapSourceFromHandle(hglobal);
                     }
-                    //  Restrict deserialization to only primitives 
-                    // and strings here to prevent potentially malicious objects from
+                    // Limit deserialization to only primitive types, which consist of the following:
+                    //
+                    // DataFormats.CommaSeparatedValue
+                    // DataFormats.FileDrop
+                    // DataFormats.Html
+                    // DataFormats.OemText
+                    // DataFormats.PenData
+                    // DataFormats.Rtf
+                    // DataFormats.Serializable
+                    // DataFormats.Text
+                    // DataFormats.UnicodeText
+                    // DataFormats.WaveAudio
+                    // DataFormats.Xaml
+                    // DataFormats.XamlPackage 
+                    // DataFormats.StringFormat
+                    // 
+                    // Out of these primitive types, we will disallow deserialization of 
```

Comment:
Do you actually *disallow* StringFormat or are you just restricting it to safe types? The PR doesn't seem to contain any "disallow" logic and the existing code just seems to use `formatter.Binder = new TypeRestrictingSerializationBinder();` so I don't quite see how that disallows `DataFormats.StringFormat`.

Might want to fix comment to be more clear what actually happens.